Guide/relationships.markdown

```diff
@@ -337,6 +337,113 @@ defining the `instance FromRow Maybe a`. This approach to joins allows
 for custom queries to leverage the autogenerated schema/IHP derived data types directly
 and cuts down on clutter from `newtype` definitions.
 
+### Multiple Nested Records
+
+`fetchRelated` can be handy to quickly fetch nested records. However, one of its limitations is that you can only go down one level. For example, if you have a `Post` with a `Comment` and that `Comment` references a  `User` (the comment author), you can't fetch the `User`.
+
+A common way for solving this would be using a custom data type, and fetching the data manually. Let's take the above example, and assume we have this schema:
+
+```
+posts:
+- id
+- title
+
+comments:
+- id
+- post_id # Reference to the `Post`
+- user_ud # Reference to the `User` that is the author.
+
+user:
+- id
+- name
+```
+
+We can define a custom data type like this under `Web/Types.hs`:
+
+```haskell
+{-| The result of fetching a `Post` with all the nested records.
+
+@see `fetchPostWithRecords`
+-}
+data PostWithRecords = PostWithRecords
+    { post :: !Post
+    , comments :: ![Comment]
+    , commentUsers :: ![User]
+    } deriving (Show)
+```
+
+And then we can fetch the data like this. Let's place this code under `Application/Helper/Controller.hs` as it might be used in multiple controllers:
+
+```haskell
+{-| a `Post` with all the nested records along with the referencing `Comment`s,
+and the authors of those comments.
+-}
+fetchPostWithRecords :: (?modelContext :: ModelContext) => Id Post -> IO PostWithRecords
+fetchPostWithRecords postId = do
+    post <- fetch postId
+
+    -- Comments referencing the post ID.
+    comments <- query @Comment
+        |> filterWhere (#postId, postId)
+        |> fetch
```

Comment:
Can also be written as

```suggestion
    comments <- fetch post.comments
```

Reply:
To confirm - will it fetch multiple comments? (if so, cool :smile: )

Reply:
yes exactly 👍  when `fetchRelated` is not called, the `comments` field is a query builder with the right `WHERE`  condition already applied (so declared like `data Post = Post { .., comments :: QueryBuilder "comments" }` )

Reply:
I tried to do it on my repo, and I get an error

https://github.com/amitaibu/ihp-landing-page/pull/13/files#r1205985723

Reply:
As `Comment` is referencing the `Post`, I don't expect to have `post.comments`

Reply:
Sorry, it's working. In my case I had to switch `landingPage.paragraphCtas` with  `landingPage.paragraphCtasRefLandingPages`

Reply:
I'll add a comment in the docs to clarify